### PR TITLE
Merge in less-sensitive updates from the project template

### PR DESCRIPTION
**Pull request by pacrob**
### What was wrong?

The project [template](https://github.com/ethereum/ethereum-python-project-template) was updated. This is to merge in many of the changes from it.

Because there are so many changes, this PR is only for files that do not touch dependency or python versions (except for adding pydocstyle). Those will be done separately.

Files that need updating in following pr:
- setup.py
- tox.ini
- mypy.ini
- .circleci/config.yml

[//]: # (See: https://py-evm.readthedocs.io/en/latest/contributing.html#pull-requests)
- [x] Add entry to the [release notes](https://github.com/ethereum/py-evm/blob/master/newsfragments/README.md)

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright 2017-2019 Ethereum Foundation
+Copyright (c) 2020 The Ethereum Foundation
```
Comment (kclowes) [inline LICENSE:3]: I have no idea how this works, but should we update this annually? As in, should this be 2023 now?

**Review by kclowes — APPROVED** [review]
Thank you for doing this! I didn't look very closely, but will this throw off the commit history between py-evm and the template next time we merge it in since you didn't take the whole template history?

File: LICENSE (hunk 1 above)
Comment (pacrob) [reply LICENSE:3]: As discussed, I'll update to the range of years in which changes were made to the library.

**Comment by kclowes** [comment]
Got it. Sounds good to me! Thanks again!

**Pull request merged**